modules/fundamental/src/advisory/endpoints/mod.rs

```diff
@@ -166,8 +171,10 @@ struct UploadParams {
 pub async fn upload(
     service: web::Data<IngestorService>,
     web::Query(UploadParams { issuer, labels }): web::Query<UploadParams>,
+    content_type: Option<web::Header<header::ContentType>>,
     bytes: web::Bytes,
 ) -> Result<impl Responder, Error> {
+    let bytes = decompress_async(bytes, content_type.map(|ct| ct.0)).await??;
```

Comment:
Just to clarify, this means the ingestor never sees the compressed file, yeah? So hashes are for the underlying document after decompression? If so, looks great.

Reply:
Correct.